# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmCode.cpp
# Commit: 80d744324b5c
# Full Hash: 80d744324b5cfca8ccadcf24d5ab61134ba17526
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2024-11-08 04:19:40
# Description:
#   Bug 1928645 - wasm: Clear padding around lazy stubs. r=jseward
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D228245
# ==============================================================================

diff -r b8237ae387f3 -r 80d744324b5c js/src/wasm/WasmCode.cpp
--- a/js/src/wasm/WasmCode.cpp	Thu Nov 07 21:28:27 2024 +0200
+++ b/js/src/wasm/WasmCode.cpp	Thu Nov 07 19:05:45 2024 +0000
@@ -608,7 +608,14 @@
     AutoMarkJitCodeWritableForThread writable;
 
     masm.executableCopy(codeStart);
-    memset(codeStart + codeLength, 0, allocationLength - codeLength);
+
+    // Clear the padding between the end of the code and the end of the
+    // allocation.
+    uint8_t* allocationEnd = allocationStart + allocationLength;
+    uint8_t* codeEnd = codeStart + codeLength;
+    MOZ_ASSERT(codeEnd <= allocationEnd);
+    size_t paddingAfterCode = allocationEnd - codeEnd;
+    memset(codeEnd, 0, paddingAfterCode);
 
     if (!stubCodeBlock->segment->linkAndMakeExecutableSubRange(
             writable, masm, allocationStart, codeStart, allocationLength)) {
